src/Databases/MysqlDatabase.php

```diff
@@ -30,7 +30,13 @@ public function setConfig(array $config) {
      * @return string
      */
     public function getDumpCommandLine($outputPath) {
-        return sprintf('mysqldump --routines --host=%s --port=%s --user=%s --password=%s %s > %s',
+    	$extras = [];
+    	if (array_key_exists('singleTransaction', $this->config) && $this->config['singleTransaction'] === true) {
+    		$extras[] = '--single-transaction';
+    	}
+    	$extra = count($extras) > 0 ? ' '. implode(' ', $extras) : '';
+    	$command = 'mysqldump --routines'.$extra.' --host=%s --port=%s --user=%s --password=%s %s > %s';
```

Comment:
Could this be?

$command = 'mysqldump --routines '.$extra.' --host=%s --port=%s --user=%s --password=%s %s > %s';

I don't see any problem with the potential of an extra space here. Do you?